tools/engine_tool/lib/src/commands/format_command.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io' as io;
+
+import 'package:path/path.dart' as p;
+
+import '../logger.dart';
+import 'command.dart';
+import 'flags.dart';
+
+// The format command implementation below works by spawning another Dart VM to
+// run the program under ci/bin/format.dart. Instead, format.dart should be
+// moved under the engine_tool package and invoked by a function call. The file
+// ci/bin/format.dart should be split up so that each of its `FormatCheckers` is
+// in a seprate file under src/formatters, and they should be unit-tested.
+
+/// The 'format' command.
+final class FormatCommand extends CommandBase {
+  /// Constructs the 'query' command.
+  FormatCommand({
+    required super.environment,
+  }) {
+    argParser
+      ..addFlag(
+        allFlag,
+        abbr: 'a',
+        help: 'By default only dirty files are checked. This flag causes all '
+              'files to be checked. (Slow)',
+        negatable: false,
+      )
+      ..addFlag(
+        fixFlag,
+        abbr: 'f',
+        help: 'Fixes formatting.',
+        defaultsTo: true,
+      )
+      ..addFlag(
+        printDiffsFlag,
+        abbr: 'p',
+        help: 'Output diffs that fix formatting errors to the logger.',
+        negatable: false,
+      )
+      ..addFlag(
+        quietFlag,
+        abbr: 'q',
+        help: 'Silences all log messages except for errors and warnings',
+        negatable: false,
+      )
+      ..addFlag(
+        verboseFlag,
+        abbr: 'v',
+        help: 'Prints verbose output',
+        negatable: false,
+      );
+  }
+
+  @override
+  String get name => 'format';
+
+  @override
+  String get description => 'Fixes formatting.';
+
+  @override
+  Future<int> run() async {
+    final bool all = argResults![allFlag]! as bool;
+    final bool fix = argResults![fixFlag]! as bool;
+    final bool printDiffs = argResults![printDiffsFlag]! as bool;
```

Comment:
Currently if I want to see the diffs without updating files I need to do:

```
et --no-fix --print-diffs
```

Would it make sense to make print diffs default to the negation of fix? And make fix default to the negation of print diffs?

For example:

```
et format                # Updates files, no diff
et format --no-fix       # Shows diff, does not update files
et format --print-diffs  # Shows diff, does not update files
```

Reply:
Why not just one flag "--dry-run" and drop "--no-fix" and "--print-diffs" ?

Reply:
Went with `--dry-run`.